haskell-indentation.el

```diff
@@ -323,7 +340,42 @@ fixes up only indentation."
            (car (haskell-indentation-first-indentation)) cursor-in-whitespace)
         (haskell-indentation-reindent-to pi cursor-in-whitespace))))))
 
-
+(defun haskell-indentation-common-electric-command (arg)
+  "Call `self-insert-command' to insert the character typed ARG times
+and indent when all of the following are true:
+
+1) The character is the first non-whitespace character on the line.
+2) There is only one possible indentation position.
+3) The variable `haskell-use-electric-keys' is non-nil.
+4) The point is not in a comment, string, or quasiquote."
+  (interactive "*p")
+  (let* ((col (haskell-indentation-current-indentation))
+         (at-indent? (= col (current-column)))
+         ind)
+    (self-insert-command arg)
+    (when (and at-indent?
+               haskell-indentation-electric-flag
```

Comment:
Thanks, I don't know why I didn't think of that. At that point `haskell-indentation-electric-flag` might as well just be used as a boolean, so I changed it accordingly.